functions/v2/firebase/upper-firestore/main.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START functions_cloudevent_firebase_reactive]
+from cloudevents.http import CloudEvent
+import functions_framework
+from google.cloud import firestore
+from google.events.cloud import firestore as firestoredata
+
+client = firestore.Client()
+
+
+# Converts strings added to /messages/{pushId}/original to uppercase
+@functions_framework.cloud_event
+def make_upper_case(cloud_event: CloudEvent) -> None:
+    firestore_payload = firestoredata.DocumentEventData()
+    firestore_payload._pb.ParseFromString(cloud_event.data)
+
+    path_parts = firestore_payload.value.name.split("/documents/")[1].split("/")
```

Comment:
This section dealing with path parts is a bit complicated. What are valid inputs for firestore_payload.value.name?  i.e. I suspect this will behave incorrectly if the document path is something like "foo/documents/foo/documents/bar" ?

I wonder if it can be simplified by making the parsing a bit more explicit i.e. something along the lines of:
```
path_parts = firestore_payload.value.name.split("/")
separator_idx = path_parts.index("documents")
collection_path = ...
document_path = ...
```
?